app/RouteHandlers/SofortNotificationHandler.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\App\RouteHandlers;
+
+use DateTime;
+use Psr\Log\LogLevel;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use WMDE\Fundraising\Frontend\Factories\FunFunFactory;
+use WMDE\Fundraising\Frontend\PaymentContext\RequestModel\SofortNotificationRequest;
+use WMDE\Fundraising\Frontend\PaymentContext\ResponseModel\SofortNotificationResponse;
+
+class SofortNotificationHandler {
+
+	private $ffFactory;
+
+	public function __construct( FunFunFactory $ffFactory ) {
+		$this->ffFactory = $ffFactory;
+	}
+
+	public function handle( Request $request ): Response {
+		$useCase = $this->ffFactory->newHandleSofortPaymentNotificationUseCase( $request->query->get( 'updateToken' ) );
+
+		$response = $useCase->handleNotification( $this->newUseCaseRequestFromRequest( $request ) );
+
+		$this->logResponseIfNeeded( $response, $request );
+
+		if ( !$response->notificationWasHandled() ) {
+			return new Response( 'Bad request', Response::HTTP_BAD_REQUEST );
+		} elseif ( $response->hasErrors() ) {
```

Comment:
Please check for Errors first, otherwise this statement will never be reached, as the error case also is unhandled.